x/protocolpool/keeper/keeper.go

```diff
@@ -90,6 +101,138 @@ func (k Keeper) GetCommunityPool(ctx context.Context) (sdk.Coins, error) {
 	return k.bankKeeper.GetAllBalances(ctx, moduleAccount.GetAddress()), nil
 }
 
+func (k Keeper) withdrawContinuousFund(ctx context.Context, recipient sdk.AccAddress) (sdk.Coin, error) {
+	cf, err := k.ContinuousFund.Get(ctx, recipient)
+	if err != nil {
+		if errors.Is(err, collections.ErrNotFound) {
+			return sdk.Coin{}, fmt.Errorf("no continuous fund found for recipient: %s", recipient.String())
+		}
+		return sdk.Coin{}, err
+	}
+	denom, err := k.iterateAndUpdateFundsDistribution(ctx)
+	if err != nil {
+		return sdk.Coin{}, fmt.Errorf("error while iterating all the continuous funds: %w", err)
+	}
+	// get allocated continuous fund
+	fundsAllocated, err := k.getAndResetDistributedFunds(ctx, recipient)
+	if err != nil {
+		return sdk.Coin{}, fmt.Errorf("error while getting distributed funds: %w", err)
+	}
+
+	recipientBal := k.bankKeeper.GetAllBalances(ctx, recipient)
```

Comment:
I don't get this, can you explain? What if they use their address for something else.

Reply:
Sorry, I didn't get your question. Could you re-frame it?

Reply:
You are getting all the balances from the recipient. If the recipient used their address for something else in the meantime (e.g received 1m ATOM from the ICF or smth), this will mess up with the calculation below.
This seems to assume a recipient address is only used for receiving funds and the funds don't move (I could remove some funds as well and never trigger MaxDistributedCapital)

Reply:
yes, there is a chance that MaxDistributedCapital check is never triggered but marko says that it shouldn't affect because the fund distribution here is continuous until cancelled.

Reply:
No, `getAndResetDistributedFunds` currently resets your available funds to 0 (https://github.com/cosmos/cosmos-sdk/pull/18471/files/c57bd5f952b189c0a403b2ae7fd72eebe81d95bd#diff-dcf594a333e20d0c216d697c5b17b3a407ca56b2f8acd2f6f1e4a13904f5a513R228-R232). And because you have too many tokens in your wallet, you'll just lose that distribution.
And this will not be clear at all for the receiver. They'll just get nothing and have no way to "claim" that back.

Imho that is a very weird to base it on what the user holds.

Reply:
oh right, seems reasonable. I'll make a quick change in resetting allocated distribution funds.

Reply:
Ok, your fix does ease one concern, thanks! I do still think `MaxDistributedCapital` is not calculated properly.

Reply:
hmm, do you have anything in mind ?
Initially I used `MaxDistributedCapital` as the capital amount to be reached and the fund distribution would stop when reached and removed that entry.
But this design has undergone various opinions and left the current way it is now.

Reply:
The limit, in my opinion, makes sense, you may not want to fund someone forever but may want more flexibility that payment tranches.
My issue is that value that is checked against `MaxDistributedCapital` doesn't actually check against the total distributed capital.
It just checks against the user balance, which may change because of external factors.

Reply:
yes, I understand your concern. 
But lets wait for marko's thoughts on this 
cc: @tac0turtle

Reply:
i hear your concern, the issue becomes how do you reset the funds. If protocolpool now keeps track of the funds sent, then a user moves funds out of the account, protocolpool now needs to know the users funds left the account, either through a query like we have here or a hook. Secondly, if the worry is i send funds to this account which means i get less tokens from the chain, it would incentivise me not to send excess funds to the account. If the user removes funds form the account this is fine as well as they would be using it for the intended purpose the fund got created for. This does mean we could never hit the max amount, if we wanted to hit the max then stop indefinitely, users would use streams which was added in a previous pr

Reply:
I do understand the idea better, but we should write down all those assumptions in the documentation. 
Especially that your usage of the address affects the amount you are getting.
If indeed, hitting the max means you are being skipped, then I guess this should be reverted as it was the intended behavior: https://github.com/cosmos/cosmos-sdk/pull/18471/commits/e0f0d2d14d42e559dd20bd683cc7a9bc2da28886